### ISSUE TITLE
the performance of NVDA falls too to enable windows defender

### ISSUE THREAD

**Issue by fernando-jose-silva** [nvaccess/nvda#6214]
Good Morning.
When using the virus before defnder windows on windows 10.0.586, to navigate folders in Windows Explorer NVDA is very slow, pressing arrow down to read the file or directory name selected NVDA takes about 2 seconds to read the item and veses or can not read the name of the selected file or folder.
By disabling windows defender performance back to be satisfactory.
I tested with the narrator and performance with windows defender enabled is not decreased so dramatically.
My machine is modest, a cori i v 1.8MHz, 4GB ran, and a hard drive of 500GB, Windows 64bit 10.0.586.
There is something that can be done in NVDA to increase their performance in windows explorer even with the windows defender enabled.
Thank you very much.

**Comment by lpintes** [comment]
This is probably not Windows-defender or antivirus-related problem. I noticed it also by using AVG Protection.
Sometimes, list items are not read by NVDA, for example when I open the network connection folder. However the list works, I can blindly select an item and press enter to see the network connectio status for example.
Sometimes after, say 10 - 20 seconds, the list works normally and smoothly. Sometimes it doesn't work at all, a NVDA restart is not enough to correct the problem.

**Comment by bhavyashah** [comment]
I can confirm similar behaviour even with NVDA version 2017.2. In an experience I had last week, on my 8 gb Ram and I5 endowed Windows 8.1 laptop, running a full scan of Windows Defender caused NVDA's CPU performance to constantly stay at levels of at least 20-25%, which is highly unusual. To put this into perspective, NVDA's processor consumption was consistently higher, according to my periodic checks, than the sum of the that taken up by the Windows Defender UI and Anti-malware executable combined. In fact, similar to what has already been mentioned, NVDA was not reading the list items of the processes in the Task manager itself, whereas Narrator was. Restarting NVDA yielded nothing as well. It is worth noting that NVDA did resume complete normalcy and functionality once the Windows Defender full scan finished, i.e. its CPU consumption dropped to well below the 10% mark and the Task manager became navigable too. I would like to urge a triage of this ticket, for such performance regressions are quite productivity-hampering.

**Comment by jcsteh** [comment]
P2 to investigate. I can understand why this might happen while the app is in focus - it's probably firing a lot of events - but this really shouldn't affect us when it's in the background.

For more technical users: I'd be interested to know if this happens with UIA disabled. Obviously, using NVDA with UIA disabled isn't really practical, since a lot of functionality is hindered or broken, but testing this would provide valuable information. To do so:

1. Open the NVDA Python console.
2. Enter the following: `import config; config.conf["UIA"]["enabled"] = False`
3. Save your NVDA configuration and restart NVDA.
4. Run the scan.
5. Observe whether the performance degradation/excessive resource consumption occurs.

To restore UIA, repeat steps 1 to 3, but change `False` in step 2 to `True`.

Technical: I'm thinking this has something to do with UIA event spam, since we listen for all UIA events.

**Comment by Adriani90** [comment]
@fernando-jose-silva how is this behavior in NVDA 2019.3.1?

**Comment by fernando-jose-silva** [comment]
I don't have this problem anymore, thank you very much.

**Comment by Adriani90** [comment]
Closing as works for me.
